src/Chapter2.hs

```diff
@@ -842,7 +857,12 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate _ [] = []
+rotate n xs
+    | n < 0 = []
+    | n == (length xs) = xs
```

Comment:
This is an interesting check, but maybe instead you can use `mod n len`, so it would be even more efficient!